Bugsnag/KSCrash/Source/KSCrash/Recording/BSG_KSCrashReport.c

```diff
@@ -926,7 +927,17 @@ void bsg_kscrw_i_writeThread(const BSG_KSCrashReportWriter *const writer,
         writer->addBooleanElement(writer, BSG_KSCrashField_Crashed,
                                   isCrashedThread);
         writer->addBooleanElement(writer, BSG_KSCrashField_CurrentThread,
-                                  thread == bsg_ksmachthread_self());
+                                  isSelfThread);
+
+        // Fetching the current thread name is only safe as of libpthread-330.201.1
+        // which was used in ios+tvos 12 and macos 10.14
+        if(__builtin_available(iOS 12.0, tvOS 12.0, macOS 10.14, *)) {
```

Comment:
`__builtin_available` is not async-safe...

some more info in this comment:

https://github.com/bugsnag/bugsnag-cocoa/blob/master/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMach.c#L66-L69